### ISSUE TITLE
Great Job

### ISSUE THREAD

**Issue by alphaDroid89** [prime-optimus/cricinfo-unofficial#1]
@prime-optimus Thanks for this wonderful api, so far i wait for this one. if possible to gave Team flag/Icon in both score and all matches api.

Thanks

**Comment by prime-optimus** [comment]
Hi @SkyNite, Thanks for the appreciation. And as far as your request is concerned, I wanted to keep this API data centric. Implementation specific details like Team Flag, Flag icon size etc. should ne the choice of developer designing interfaces.

However, i am keeping this ticket open for some time. If others support your cause, I'll be more than happy to incorporate this change.

Thanks again.

**Comment by maxakash** [comment]
Well, that would be nice if add that feature too.

**Comment by dragonlord2025** [comment]
+1 for the API @prime-optimus . Yeah, This will reduce the overhead for the developers to find flag's and icon's of teams and add it. It's a good to have feature. 

What about licence? is it under apache licence right? 
Add licence file.

**Comment by prime-optimus** [comment]
Thanks for the suggestion @shivarajp . 

Yes it's under Apache license. I have added the license file too.

**Comment by dragonlord2025** [comment]
Cool..

**Comment by nitish29** [comment]
Great job on the Match API! When do you plan on releasing the player API as well?

**Comment by prime-optimus** [comment]
Thanks nitish29, I'm currently overwhelmed with some other projects. I'll be working on the Player API as soon as I get some time.

**Comment by dragonlord2025** [comment]
@MukeshPrasad  cricinfo provides these feeds for free? What if there are we
hit 1000's of hits they will block it?

On Tue, Apr 5, 2016 at 12:07 AM, Mukesh Prasad notifications@github.com
wrote:

> Thanks nitish29, I'm currently overwhelmed with some other projects. I'll
> be working on the Player API as soon as I get some time.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/prime-optimus/cricinfo-unofficial/issues/1#issuecomment-205440908

## 

Shivaraj Patil |  Triveous |  shivrajp130@gmail.com urbaddy@gmail.com |
 +91-7353489699

**Comment by prime-optimus** [comment]
That's where caching jumps in. If all the 1000 requests are concurrent will be having same data so calls to cricinfo will be only 1.

**Comment by prime-optimus** [comment]
That's where caching comes into the picture. Most of the requests will be served from within the cache and cricinfo will be contacted for the least amount of time, i.e. when new data is available.

**Comment by dragonlord2025** [comment]
Yeah, agreed. Any idea till how many request's/day/month cricinfo will
allow for free? have you ever experienced such a scenario or have any
stats? what is your view point on this? This is very important for me if I
integrate this and later for some reason cricinfo blocked means gone I will
loose all users.

Any help would be really helpful.

Thanks,

**Comment by prime-optimus** [comment]
I'm not using any standard Cricinfo API, rather I am scraping data from Cricinfo's HTML website. And I don't think(or know of) any limitations in accessing the Cricinfo site. 

So while integrating this API I will not fear of any blocking by Cricinfo.

**Comment by dragonlord2025** [comment]
Hi Mukesh,

Thanks for your reply.

I think they will block your server IP if it is hitting few thousand
requests everyday. Anyway nice work man.

 Thanks a lot.

On Tue, Apr 5, 2016 at 8:38 PM, Mukesh Prasad notifications@github.com
wrote:

> I'm not using any standard Cricinfo API, rather I am scraping data from
> Cricinfo's HTML website. And I don't think(or know of) any limitations in
> accessing the Cricinfo site.
> 
> So while integrating this API I will not fear of any blocking by Cricinfo.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/prime-optimus/cricinfo-unofficial/issues/1#issuecomment-205850121

## 

Shivaraj Patil |  Triveous |  shivrajp130@gmail.com urbaddy@gmail.com |
 +91-7353489699

**Comment by prime-optimus** [comment]
Yeah, there is a faint possibility. I've tested it by hitting cricinfo multiple times using 200 threads. But they haven't blocked me out. I don't think we can generate more traffic to cricinfo than it already is handling.

**Comment by dragonlord2025** [comment]
Yeah, but still we have to make sure that we are not crossing the limit.

On Wed, Apr 6, 2016 at 12:04 AM, Mukesh Prasad notifications@github.com
wrote:

> Yeah, there is a faint possibility. I've tested it by hitting cricinfo
> multiple times using 200 threads. But they haven't blocked me out. I don't
> think we can generate more traffic to cricinfo than it already is handling.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/prime-optimus/cricinfo-unofficial/issues/1#issuecomment-205937489

## 

Shivaraj Patil |  Triveous |  shivrajp130@gmail.com urbaddy@gmail.com |
 +91-7353489699

**Comment by prime-optimus** [comment]
Yup, sure. But only if know if there is any limit.

**Comment by dragonlord2025** [comment]
There is a limit. This happened with my frieds, we are free to use until
they notice.

On Wed, Apr 6, 2016 at 12:12 AM, Mukesh Prasad notifications@github.com
wrote:

> Yup, sure. But only if know if there is any limit.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/prime-optimus/cricinfo-unofficial/issues/1#issuecomment-205942069

## 

Shivaraj Patil |  Triveous |  shivrajp130@gmail.com urbaddy@gmail.com |
 +91-7353489699

**Comment by prime-optimus** [comment]
Yeah, but there's a risk worth taking if you don't want to use any Paid Cricket API.

**Comment by dragonlord2025** [comment]
Ha ha Exactly.

On Thu, Apr 7, 2016 at 1:17 AM, Mukesh Prasad notifications@github.com
wrote:

> Yeah, but there's a risk worth taking if you don't want to use any Paid
> Cricket API.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/prime-optimus/cricinfo-unofficial/issues/1#issuecomment-206529844

## 

Shivaraj Patil |  Triveous |  shivrajp130@gmail.com urbaddy@gmail.com |
 +91-7353489699

**Comment by karneaud** [comment]
+1

**Comment by ahripon107** [comment]
score is not updating since yesterday.. please check it bro..

**Comment by prime-optimus** [comment]
I think it is in accordance to RSS feed provided by Cricinfo. Check here.

http://static.espncricinfo.com/rss/livescores.xml

**Comment by ahripon107** [comment]
Both the apis are getting 502 or 503. Think it is blocked by cricinfo.

**Comment by prime-optimus** [comment]
I checked them. It is working now.
